lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/AccessControlRulesRestExtractor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.inject.Inject;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.StringResponseHandler;
+import io.trino.plugin.base.util.JsonUtils;
+
+import java.net.URI;
+
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class AccessControlRulesRestExtractor
+{
+    private final HttpClient httpClient;
+    private final RestBasedAccessControlConfig config;
+
+    @Inject
+    public AccessControlRulesRestExtractor(@ForAccessControlRules HttpClient httpClient, RestBasedAccessControlConfig config)
+    {
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        this.config = requireNonNull(config, "config is null");
```

Comment:
Instead of having a field for config - can we use field for `URI`

Reply:
And a field for `jsonPointer`

Reply:
hmm don't think so. config is bound in a specific RestAccessControlModule and can be injected here easily. If you want to inject URI or jsonPointer you would have to extract them from config and bind to the context. I don't like fetching config in module configure method. Unless it can be achieved other way? WDYT?

Reply:
The constructor could have this config, it could be something like this 

```
    public AccessControlRulesRestExtractor(@ForAccessControlRules HttpClient httpClient, RestBasedAccessControlConfig config)
    {
        this.httpClient = requireNonNull(httpClient, "httpClient is null");
        requireNonNull(config. "config is null");
        this.uri  = config.getUri();
        this.pointer = config.getJsonPointer();
     }
     
Config objects are mutable so it is not recommended to use them as fields in objects.

Reply:
> The constructor could have this config, it could be something like this
> 
> ```
>     public AccessControlRulesRestExtractor(@ForAccessControlRules HttpClient httpClient, RestBasedAccessControlConfig config)
>     {
>         this.httpClient = requireNonNull(httpClient, "httpClient is null");
>         requireNonNull(config. "config is null");
>         this.uri  = config.getUri();
>         this.pointer = config.getJsonPointer();
>      }
>      
> Config objects are mutable so it is not recommended to use them as fields in objects. 
> ```

ok. that is perfectly doable. will do

Reply:
refactored

Reply:
regarding <dependency>
            <groupId>io.airlift</groupId>
            <artifactId>node</artifactId>
            <scope>test</scope>
        </dependency>

It is not explicit. But they are required by some tests. If I remove it the build fails